sympy/core/power.py

```diff
@@ -1180,6 +1180,12 @@ def _eval_is_polynomial(self, syms):
             return True
 
     def _eval_is_rational(self):
+        # The evaluation of self.func below can be very expensive in the case
+        # of integer**integer if the exponent is large.  We should try to exit
+        # before that if possible:
+        if (self.exp.is_integer and self.exp.is_nonnegative
+                and self.base.is_rational):
```

Comment:
Does it really have to be non-negative? Should be enough with integer exponent and rational base.

Reply:
And base nonzero if exponent is negative.

Reply:
I either need to check that the exponent is nonnegative or that the base is nonzero. Perhaps best to do both...